### PR TITLE
Some improvements to docs.

### PR DESCRIPTION
**Pull request by jlewi**
* Remove note about building ksonnet from source; there is now a released artifact for version 0.8.0

* Remove instructions about setting up K8s clusters; just link to the K8s docs.

* Move notes about minikube and GKE to a troubleshooting section.

* Add info about why ksonnet.

### CODE REVIEW

**Comment by jlewi** [comment]
@nkashy1 ptal

File: README.md [hunk 1]
```diff
@@ -76,6 +61,30 @@ effort in a portable fashion between different environments.
...
+## Troubleshooting
+
+### Minikube
+
+On [Minikube](https://github.com/kubernetes/minikube) the Virtualbox/VMware drivers for Minikube are recommended as there is a known
+issue between the KVM/KVM2 driver and TensorFlow Serving. The issue is tracked in [kubernetes/minikube#2377](https://github.com/kubernetes/minikube/issues/2377).
+
+### RBAC clusters
+
+If you are running on a K8s cluster with RBAC enabled, you may get an error like the following when deploying Kubeflow:
```
Comment (foxish) [inline README.md]: Should link to https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control and https://kubernetes.io/docs/admin/authorization/rbac/#command-line-utilities.

File: README.md [hunk 2]
```diff
@@ -21,27 +21,12 @@ Ultimately, we want to have a set of simple manifests that give you an easy to u
...
+This documentation assumes you have a Kubernetes cluster already available. 
 
-### Minikube
-
-[Minikube](https://github.com/kubernetes/minikube) is a tool that makes it easy to run Kubernetes locally. Minikube runs a
-single-node Kubernetes cluster inside a VM on your laptop for users looking to try out Kubernetes or develop with it day-to-day.
-
-The below steps apply to a minikube cluster - the latest version as of writing this documentation is 0.23.0. You must also have
-kubectl configured to access minikube.
+If you need help setting up a Kubernetes cluster please refer to [Kubernetes' Setup](https://kubernetes.io/docs/setup/).
```
Comment (foxish) [inline README.md]: Remove ' after Kubernetes.

File: README.md (hunk 1 above)
Comment (jlewi) [reply README.md]: Done

**Pull request merged**